src/datasources/cache/cache.first.data.source.ts

```diff
@@ -47,19 +45,15 @@ export class CacheFirstDataSource {
     params?: NetworkRequest,
     expireTimeSeconds?: number,
   ): Promise<T> {
-    try {
-      const cached = await this.cacheService.get(key, field);
-      if (cached != null) {
-        this.logger.debug(`[Cache] Cache hit: ${key}`);
-        return JSON.parse(cached);
-      }
-      this.logger.debug(`[Cache] Cache miss: ${key}`);
-      const { data } = await this.networkService.get(url, params);
-      const rawJson = JSON.stringify(data);
-      await this.cacheService.set(key, field, rawJson, expireTimeSeconds);
-      return data;
-    } catch (error) {
-      throw this.httpErrorFactory.from(error);
+    const cached = await this.cacheService.get(key, field);
+    if (cached != null) {
+      this.logger.debug(`[Cache] Cache hit: ${key}`);
+      return JSON.parse(cached);
     }
+    this.logger.debug(`[Cache] Cache miss: ${key}`);
+    const { data } = await this.networkService.get(url, params);
+    const rawJson = JSON.stringify(data);
+    await this.cacheService.set(key, field, rawJson, expireTimeSeconds);
+    return data;
```

Comment:
Removing here the throw exception?

Reply:
The reason is in the PR description:

> CacheFirstDataSource no longer maps error coming from the network layer
The reason behind this decision is that the CacheFirstDataSource should not make assumptions if the consumers want the error mapped or not.
The error mapping would keep the status code and message if they were available in the payload however this decision should be done on each data source method (exchange api methods for example do not forward the error the status code or message).